src/activations.jl

```diff
@@ -84,7 +84,7 @@ leakyrelu(x, a=oftf(x, 0.01)) = max(a * x, x)
 activation function capped at 6.
 See [Convolutional Deep Belief Networks on CIFAR-10](https://www.cs.toronto.edu/~kriz/conv-cifar10-aug2010.pdf)
 """
-relu6(x) = min(relu(x), oftype(x, 6))
+relu6(x) = clamp(x, Int8(0), Int8(6))  # (small types preserve Int32, not really necc.)
```

Comment:
```suggestion
relu6(x) = clamp(x, oftype(x,0), oftype(x, 6))
```
? seems a bit less arbitrary. Not a big deal though

Reply:
Can do, maybe that's less weird to read. They compile to the same thing:
```
julia> relu6(x) = clamp(x, Int8(0), Int8(6))
relu6 (generic function with 1 method)

julia> @code_native relu6(1f0)
	.section	__TEXT,__text,regular,pure_instructions
; ┌ @ REPL[54]:1 within `relu6`
	xorps	%xmm2, %xmm2
; │┌ @ math.jl:67 within `clamp`
	maxss	%xmm0, %xmm2
	movabsq	$6549393552, %rax               ## imm = 0x1865FD090
	movss	(%rax), %xmm3                   ## xmm3 = mem[0],zero,zero,zero
	movaps	%xmm3, %xmm1
	cmpnltss	%xmm0, %xmm1
	andps	%xmm1, %xmm2
	andnps	%xmm3, %xmm1
	orps	%xmm2, %xmm1
; │└
	movaps	%xmm1, %xmm0
	retq
	nopw	(%rax,%rax)
; └

julia> relu6(x) = clamp(x, oftype(x,0), oftype(x, 6))
relu6 (generic function with 1 method)

julia> @code_native relu6(1f0)
	.section	__TEXT,__text,regular,pure_instructions
; ┌ @ REPL[56]:1 within `relu6`
	xorps	%xmm2, %xmm2
; │┌ @ math.jl:67 within `clamp`
	maxss	%xmm0, %xmm2
	movabsq	$6549393952, %rax               ## imm = 0x1865FD220
	movss	(%rax), %xmm3                   ## xmm3 = mem[0],zero,zero,zero
	movaps	%xmm3, %xmm1
	cmpnltss	%xmm0, %xmm1
	andps	%xmm1, %xmm2
	andnps	%xmm3, %xmm1
	orps	%xmm2, %xmm1
; │└
	movaps	%xmm1, %xmm0
	retq
	nopw	(%rax,%rax)
; └

julia> relu6(x) = min(relu(x), oftype(x, 6))  # old
relu6 (generic function with 1 method)

julia> @code_native relu6(1f0)
	.section	__TEXT,__text,regular,pure_instructions
; ┌ @ REPL[58]:1 within `relu6`
	pushq	%r15
	pushq	%r14
	pushq	%r12
	pushq	%rbx
	subq	$72, %rsp
	movss	%xmm0, 12(%rsp)
	xorps	%xmm0, %xmm0
	movaps	%xmm0, 32(%rsp)
	movaps	%xmm0, 16(%rsp)
	movabsq	$pthread_getspecific, %rax
	movl	$261, %edi                      ## imm = 0x105
	callq	*%rax
	movq	%rax, %r14
	movq	$8, 16(%rsp)
	movq	(%rax), %rax
	movq	%rax, 24(%rsp)
	leaq	16(%rsp), %rax
	movq	%rax, (%r14)
	movabsq	$6549394352, %rbx               ## imm = 0x1865FD3B0
	movq	(%rbx), %rax
	testq	%rax, %rax
	jne	L125
; │ @ REPL[58] within `relu6`
	movabsq	$jl_system_image_data, %rdi
; │ @ REPL[58]:1 within `relu6`
	leaq	1364100320(%rdi), %rsi
	movabsq	$jl_get_binding_or_error, %rax
	callq	*%rax
	movq	%rax, (%rbx)
L125:
	movq	8(%rax), %rbx
	testq	%rbx, %rbx
	je	L266
; │ @ REPL[58] within `relu6`
	movq	%rbx, 40(%rsp)
; │ @ REPL[58]:1 within `relu6`
	movabsq	$jl_box_float32, %rax
	movss	12(%rsp), %xmm0                 ## xmm0 = mem[0],zero,zero,zero
	callq	*%rax
	movq	%rax, 32(%rsp)
	movq	%rax, 56(%rsp)
	movabsq	$jl_apply_generic, %r12
	leaq	56(%rsp), %r15
	movq	%rbx, %rdi
	movq	%r15, %rsi
	movl	$1, %edx
	callq	*%r12
	movq	%rax, 32(%rsp)
	movq	%rax, 56(%rsp)
	movabsq	$4715096144, %rax               ## imm = 0x1190AA850
	movq	%rax, 64(%rsp)
	movabsq	$jl_system_image_data, %rdi
	movq	%r15, %rsi
	movl	$2, %edx
	callq	*%r12
	movq	24(%rsp), %rcx
	movq	%rcx, (%r14)
	addq	$72, %rsp
	popq	%rbx
	popq	%r12
	popq	%r14
	popq	%r15
	retq
L266:
	movabsq	$jl_undefined_var_error, %rax
	movabsq	$6115942736, %rdi               ## imm = 0x16C89E150
	callq	*%rax
; └

julia> 
```

Reply:
```suggestion
relu6(x) = clamp(x, oftype(x,0), oftype(x, 6))  # clamp promotes, but clamp(x, 0, 6) would promote x::Int32
```